src/liballoc/collections/btree/set.rs

```diff
@@ -163,18 +237,17 @@ impl<T: fmt::Debug> fmt::Debug for Difference<'_, T> {
 /// [`BTreeSet`]: struct.BTreeSet.html
 /// [`symmetric_difference`]: struct.BTreeSet.html#method.symmetric_difference
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct SymmetricDifference<'a, T: 'a> {
-    a: Peekable<Iter<'a, T>>,
-    b: Peekable<Iter<'a, T>>,
-}
+pub struct SymmetricDifference<'a, T: 'a>(MergeIterInner<Iter<'a, T>>);
 
 #[stable(feature = "collection_debug", since = "1.17.0")]
 impl<T: fmt::Debug> fmt::Debug for SymmetricDifference<'_, T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_tuple("SymmetricDifference")
-         .field(&self.a)
-         .field(&self.b)
-         .finish()
+            .field(&self.0.a.head)
+            .field(&self.0.a.tail)
+            .field(&self.0.b.head)
+            .field(&self.0.b.tail)
```

Comment:
This will look messier, but I'm not sure if we have any reason to dress up the internals in any way, it should serve us best to just debug print what we have. That would be an argument for printing this as `.field(&self.0)` or `.field(&self.0.a).field(&self.0.b)` though, and giving Peeking a debug impl.

Reply:
The remaining `fmt::Debug` for Iter uses the older alignment style here (aligning methods to the `.`). Best to keep the style consistent in the file (keep the style as it was).

Reply:
I never understood that part, just stole the debug implementation from some other liballoc struct. I guess part of the reason is to allow the "#[stable()" thingy.

I'm now finally able to produce examples of this output, like:

`Intersection(Iter([1, 2, 3]), Iter([1, 2, 3]))`

If I simply stick a `#[derive(Debug)]` onto IntersectionInner, and simplify Intersection::fmt all the way down to `f.debug_tuple("Intersection").field(&self.inner).finish()`, the output is much more clearly:

`Intersection(Stitch { a: Iter([1, 2, 3]), b: Iter([1, 2, 3]) })`